addons/spreadsheet/static/src/data_sources/data_sources.js

```diff
@@ -91,6 +102,9 @@ export class DataSources extends EventBus {
      * request a re-evaluation.
      */
     notify() {
+        if (this.pendingPromises.size !== 0) {
+            return;
+        }
```

Comment:
It seems extreme to me. If you have many RPC triggered and most of them are finished, it will still not trigger.
We should probably limit the number of triggers, like no more than once every 10 seconds, and not if you have more than X (2 maybe) waiting...
What do you think ?

Reply:
I pushed something (after you approved). The initial implementation was too naive